pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -0,0 +1,793 @@
+package awskms
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/andres-erbsen/clock"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/aws/aws-sdk-go-v2/service/kms/types"
+	"github.com/gofrs/uuid"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/pkg/server/plugin/keymanager"
+	"github.com/spiffe/spire/proto/spire/common/plugin"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	pluginName              = "aws_kms"
+	aliasPrefix             = "alias/"
+	refreshAliasesFrequency = time.Hour * 6
+	disposeAliasesFrequency = time.Hour * 24
+	aliasThreshold          = time.Hour * 48
+	disposeKeysFrequency    = time.Hour * 48
+	keyThreshold            = time.Hour * 48
+	keyArnTag               = "key_arn"
+	aliasNameTag            = "alias_name"
+	reasonTag               = "reason"
+)
+
+func BuiltIn() catalog.Plugin {
+	return builtin(New())
+}
+
+func builtin(p *Plugin) catalog.Plugin {
+	return catalog.MakePlugin(pluginName, keymanager.PluginServer(p))
+}
+
+type keyEntry struct {
+	Arn       string
+	AliasName string
+	PublicKey *keymanager.PublicKey
+}
+
+type pluginHooks struct {
+	newClient func(ctx context.Context, config *Config) (kmsClient, error)
+	// just for testing scheduleDeleteTask
+	deleteSignal chan struct{}
+}
+
+// Plugin is the main representation of this keymanager plugin
+type Plugin struct {
+	keymanager.UnsafeKeyManagerServer
+	log            hclog.Logger
+	mu             sync.RWMutex
+	entries        map[string]keyEntry
+	kmsClient      kmsClient
+	trustDomain    string
+	serverID       string
+	scheduleDelete chan string
+	cancelTasks    context.CancelFunc
+	clk            clock.Clock
+	hooks          pluginHooks
+}
+
+// Config provides configuration context for the plugin
+type Config struct {
+	AccessKeyID     string `hcl:"access_key_id" json:"access_key_id"`
+	SecretAccessKey string `hcl:"secret_access_key" json:"secret_access_key"`
+	Region          string `hcl:"region" json:"region"`
+	ServerIDPath    string `hcl:"server_id_path" json:"server_id_path"`
```

Comment:
Looking at this from an operator standpoint, the name `server_id_path` may be confusing because I'm not sure to what "path" is referring to.
I think that `server_id_file_path` would provide more clarity. For me at least, it was not obvious that this would write to a file path specified here.

Reply:
The fact that the docs are not up to date yet didn't help, but I believe you are right. I will change the name to be more descriptive as you suggested.